hudi-client/hudi-java-client/src/main/java/org/apache/hudi/execution/bulkinsert/JavaGlobalSortPartitioner.java

```diff
@@ -31,12 +32,21 @@
  *
  * @param <T> HoodieRecordPayload type
  */
-public class JavaGlobalSortPartitioner<T>
-    implements BulkInsertPartitioner<List<HoodieRecord<T>>> {
+public class JavaGlobalSortPartitioner<T> implements BulkInsertPartitioner<List<HoodieRecord<T>>> {
+
+  public JavaGlobalSortPartitioner() {
+  }
+
+  /**
+   * Constructor to create as UserDefinedBulkInsertPartitioner class via reflection
+   * @param config HoodieWriteConfig
```

Comment:
Can you give an example how this partitioner got instantiated and customized?

Reply:
This partitioner will be instantiated when user define write config property `hoodie.bulkinsert.user.defined.partitioner.class=org.apache.hudi.execution.bulkinsert.JavaGlobalSortPartitioner`. This constructor will be called via reflection in DataSourceUtils class: `createUserDefinedBulkInsertPartitioner(HoodieWriteConfig config)` and `createUserDefinedBulkInsertPartitionerWithRows(HoodieWriteConfig config)`.  
There is nothing to customize in this JavaGlobalSortPartitioner, but, for example, provided writeConfig is used for customization  of RowSpatialCurveSortPartitioner.

Reply:
I got confused because the "customized" `HoodieWriteConfig` does not really play a role here and it is ignored?

Reply:
Yes, in this case HoodieWriteConfig is ignored just because this Partitioner is not configurable at all, but it does not mean that it should not be used as `UserDefinedBulkInsertPartitioner`.  
So I think, the purpose of this task is not to make all BulkInsertPartitioners customizable with HoodieWriteConfig, but only to make them instantiable via reflection with already existing common approach for UserDefinedBulkInsertPartitioner (constructor with HoodieWriteConfig as the only parameter).  
@nsivabalan am I right?

Reply:
> Yes, in this case HoodieWriteConfig is ignored just because this Partitioner is not configurable at all, but it does not mean that it should not be used as UserDefinedBulkInsertPartitioner

That does not make sense for me.

Reply:
Before this fix:  
if user wants to use JavaGlobalSortPartitioner and he set `hoodie.bulkinsert.user.defined.partitioner.class=org.apache.hudi.execution.bulkinsert.JavaGlobalSortPartitioner`, it will not work because this partitioner could not be instantiated via reflection (as it has no constructor with writeConfig parameter).
We create this constructor to add ability to use JavaGlobalSortPartitioner as user defined partitioner just by setting it's class name in writeConfig.  

Don't know how to explain more clear. Let's wait for the author's reply.

Reply:
i think, yes it does. before the fix this partitioner had only default constructor, it was enough for it's instantiation and working, as it has no referencies to any writeconfig params

Reply:
@danny0405 can you please assign this PR to @nsivabalan to summon him to this discussion?